### PR TITLE
Refactor coefficient writing

### PR DESCRIPTION
**Pull request by KyleSiefring**
Improves performance slightly.

### CODE REVIEW

File: src/context.rs [hunk 1]
```diff
@@ -4195,12 +4137,8 @@ impl<'a> ContextWriter<'a> {
       }
       // save extra golomb codes for separate loop
       if level > (COEFF_BASE_RANGE + NUM_BASE_LEVELS) as u32 {
```
Comment (KyleSiefring) [inline src/context.rs]: same here

File: src/context.rs [hunk 2]
```diff
@@ -4140,16 +4084,14 @@ impl<'a> ContextWriter<'a> {
       }
 
       if level > NUM_BASE_LEVELS as u32 {
```
Comment (KyleSiefring) [inline src/context.rs]: level > ... as u32 then convert level to u16. This has code smell.

File: src/context.rs [hunk 3]
```diff
@@ -3827,51 +3825,42 @@ impl<'a> ContextWriter<'a> {
...
-        //   if (row < 2) return 11 + ctx;
-        // } else if (width > height) {
-        //   if (col < 2) return 16 + ctx;
-        // }
-        // if (row + col < 2) return ctx + 1;
-        // if (row + col < 4) return 5 + ctx + 1;
-        // return 21 + ctx;
-        ctx
-          + av1_nz_map_ctx_offset[tx_size as usize][cmp::min(row, 4)]
+    let ctx = ((stats + 1) >> 1).min(4);
```
Comment (tdaede) [inline src/context.rs]: any reason not to combine these into one statement

File: src/context.rs (hunk 2 above)
Comment (tdaede) [reply src/context.rs]: I think NUM_BASE_LEVELS was a #define in libaom so changing its type to u16 would also be a legit fix

Comment (tdaede) [reply src/context.rs]: actually no we actually do the conversion at v.abs() as u16 right above. I think we should just drop all those conversions

**Comment by KyleSiefring** [comment]
Closing till I fix the bug

**Pull request closed without merging**